lib/taurus/core/util/fqdn.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+#############################################################################
+##
+# This file is part of Taurus
+##
+# http://taurus-scada.org
+##
+# Copyright 2011 CELLS / ALBA Synchrotron, Bellaterra, Spain
+##
+# Taurus is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+##
+# Taurus is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+##
+# You should have received a copy of the GNU Lesser General Public License
+# along with Taurus.  If not, see <http://www.gnu.org/licenses/>.
+##
+#############################################################################
+"""
+This module contains provides the fqdn_no_alias function.
+"""
+
+import socket
+from taurus import warning
+
+
+def fqdn_no_alias(hostname):
+    """
+    Normalize hostname so that it is a full name (including domain name).
+
+    .. note:: this is similar to socket.getfqdn, but it avoids translating
+              aliases into their "real" address.
+    """
+    if hostname == "localhost" or "." in hostname:
+        # optimization: leave localhost or names including domain as they are
+        return hostname
+
+    try:
+        real, aliases, _ = socket.gethostbyname_ex(hostname)
+    except (socket.gaierror, socket.herror) as e:
+        # return the given hostname in case of error
+        warning("fqdn_no_alias: problem resolving %s: %r", hostname, e)
```

Comment:
Testing a not reachable attribute with taurus. This warning is reported many times. Is it necessary? Should we control it? The old implementation delegated it to tango (taurus layer).

```
 import taurus
 taurus.Attribute('//foo:10000/a/b/c/d')
   
MainThread     INFO     2019-09-05 15:26:01,149 TaurusRootLogger: Using PyQt4 (v4.11.4 with Qt 4.8.7 and Python 2.7.13)

MainThread     INFO     2019-09-05 15:26:01,196 taurus.qt.qtgui.icon.icon: Setting Tango icon theme (from /homelocal/cfalcon/tools/taurus.git/lib/taurus/qt/qtgui/icon/)
MainThread     INFO     2019-09-05 15:26:01,482 TaurusRootLogger: Plugin "taurus_pyqtgraph" loaded as "taurus.qt.qtgui.tpg"
MainThread     WARNING  2019-09-05 15:26:01,520 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,521 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,522 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,524 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,524 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,524 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,527 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,527 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,527 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,527 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,527 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,527 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,528 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,528 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,528 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,528 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,528 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,528 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,529 a/b/c: Could not create HW object: TRANSIENT CORBA system exception: TRANSIENT_ConnectFailed
MainThread     WARNING  2019-09-05 15:26:01,614 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,614 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,614 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,614 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,614 TaurusRootLogger: fqdn_no_alias: problem resolving foo: gaierror(-2, 'Name or service not known')
MainThread     WARNING  2019-09-05 15:26:01,615 a/b/c: Could not create HW object: TRANSIENT CORBA system exception: TRANSIENT_ConnectFailed

```

Reply:
I agree: it is better to remove it. I'll commit the change